### DISCUSSION TITLE
Redirection of Console.WriteLine to ITestOutputHelper WriteLine

### DISCUSSION THREAD

**Discussion by sjanisz** [xunit/xunit#2649] [Feedback Request]
I will copy paste my question which I have asked on SO (https://stackoverflow.com/questions/75172528/xunit-redirection-of-console-writeline-to-itestoutputhelper-writeline):

I have problem with solution from this answer: https://stackoverflow.com/a/47529356/8611327

The problem is that when I use parallel testcases execution Console.WriteLine is common for all running test instances and logging is not assigned to specific testcase, so I see logs written under last testcase that have instantiated WriteLineConverter and set it as Console output:

WriteLineConverter writeLineConverter = new WriteLineConverter(_output);
Console.SetOut(writeLineConverter);
I need to use Console.WriteLine because I have a lot of nested classes like services, then access to database, sending http requests etc where I am logging timestamps and requests, I need to be able to connect theses logs with specific testcases because in parallel scenario logs from testcases are mixed.

Eventually is it possible to somehow run tests so each testcase has its separate stack of variables rather than shared stack used by all testcases?

**Comment by bradwilson** [comment]
Yes, that's exactly why we don't capture Console anymore: https://xunit.net/docs/capturing-output

> Eventually is it possible to somehow run tests so each testcase has its separate stack of variables rather than shared stack used by all testcases?

This has to do with Console being a shared, static resource. There's no way to allocate a separate static Console for each test.
